### DISCUSSION TITLE
Automatic switching on or off of generators

### DISCUSSION THREAD

**Discussion by Crueda96** [e2nIEE/pandapower#1234] [Q&A]
There is some way to automatically turn generators on / off according to the convergence of the flow, that is, if there is a lot of generation for the current load, to be able to disconnect certain generation units; or if there is a generation deficit to supply the load to be able to incorporate generation units?

**Comment by rbolgaryn** [comment]
Hi @Crueda96 ,

you can write a controller that sets the generators on or off, based on load flow results or a preliminary analysis in the time_step

Roman
